### PR TITLE
Improve type situation in Recipe subsystem

### PR DESCRIPTION
**Pull request by sjmiles**

### CODE REVIEW

File: pkg/ts/recipe/Chef.ts [hunk 1]
```diff
@@ -12,49 +12,49 @@ import {Runtime} from '../Runtime.js';
...
       log.error('`arc` must be an Arc, not a Promise. Make sure `boostrapArc` is awaited.');
       return;
     }
     //log.groupCollapsed('executing recipe...', recipe.$meta);
-    log('---> executing recipe: ', recipe.$meta);
+    log('|-->...| executing recipe: ', recipe.$meta);
     const plan = new Parser(recipe);
     // `store` preparation
-    await StoreCook.execute(runtime, arc, plan);
+    await StoreCook.execute(runtime, arc, plan.stores);
```
Comment (Cypher1) [inline pkg/ts/recipe/Chef.ts:29]: Nice

File: pkg/ts/utils/object.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export function deepCopy<T>(datum: T): T {
   if (!datum) {
     return datum;
   } else if (Array.isArray(datum)) {
-    // This is trivially type safe but tsc cannot prove it so we have to 'promise'.
+    // This is trivially type safe but tsc needs help
```
Comment (Cypher1) [inline pkg/ts/utils/object.ts:70]: Not sure what the problem was here, unless it was too cheeky :)

File: pkg/ts/utils/object.ts [hunk 3]
```diff
@@ -78,7 +78,7 @@ export function deepCopy<T>(datum: T): T {
   } else {
     return datum;
   }
-};
+}
```
Comment (Cypher1) [inline pkg/ts/utils/object.ts:81]: Whoops, bad Rust habit. Thanks for fixing that

**Review by Cypher1 — COMMENTED** [review]
LGTM

File: pkg/ts/utils/object.ts (hunk 2 above)
Comment (sjmiles) [reply pkg/ts/utils/object.ts:70]: No, I just stumbled over it because I started looking for Promise.

File: pkg/ts/utils/object.ts (hunk 3 above)
Comment (sjmiles) [reply pkg/ts/utils/object.ts:81]: NBD, I just updated the lint to highlight semis.

**Review by Cypher1 — COMMENTED** [review]
LGTM

**Pull request merged**